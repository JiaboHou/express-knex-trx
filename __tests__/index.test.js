const yo = require('../src').default;

describe('test test', () => {
  it('returns "hi"', () => {
    expect(yo(true)).toEqual('hi');
  });

  it('return "bye"', () => {
    expect(yo(false)).toEqual('bye');
  });
});
