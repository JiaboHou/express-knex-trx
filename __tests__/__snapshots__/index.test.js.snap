// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/index knexTrxErrorHandler failed rollback passes the expected error message to the next middleware 1`] = `"An error occurred while trying to perform a knex trx rollback due to a previous error: test error"`;

exports[`src/index knexTrxErrorHandler when trx is not an instance of Knex.Transaction throws an Error when res.locals.trx is a boolean 1`] = `[Error: An error occurred while trying to perform a knex trx rollback due to a previous error: ]`;

exports[`src/index knexTrxErrorHandler when trx is not an instance of Knex.Transaction throws an Error when res.locals.trx is a function without a "rollback" member 1`] = `[Error: An error occurred while trying to perform a knex trx rollback due to a previous error: ]`;

exports[`src/index knexTrxErrorHandler when trx is not an instance of Knex.Transaction throws an Error when res.locals.trx is a number 1`] = `[Error: An error occurred while trying to perform a knex trx rollback due to a previous error: ]`;

exports[`src/index knexTrxErrorHandler when trx is not an instance of Knex.Transaction throws an Error when res.locals.trx is a string 1`] = `[Error: An error occurred while trying to perform a knex trx rollback due to a previous error: ]`;

exports[`src/index knexTrxErrorHandler when trx is not an instance of Knex.Transaction throws an Error when res.locals.trx is an array 1`] = `[Error: An error occurred while trying to perform a knex trx rollback due to a previous error: ]`;

exports[`src/index knexTrxErrorHandler when trx is not an instance of Knex.Transaction throws an Error when res.locals.trx is an object 1`] = `[Error: An error occurred while trying to perform a knex trx rollback due to a previous error: ]`;

exports[`src/index knexTrxErrorHandler when trx is not an instance of Knex.Transaction throws an Error when res.locals.trx is null 1`] = `[Error: An error occurred while trying to perform a knex trx rollback due to a previous error: ]`;

exports[`src/index knexTrxErrorHandler when trx is not an instance of Knex.Transaction throws an Error when res.locals.trx is undefined 1`] = `[Error: An error occurred while trying to perform a knex trx rollback due to a previous error: ]`;
